Add `OrderedDict.fromkeys`

AlexWaygood opened the pull request:
Closes #3800 (no other solution possible until we have higher-kinded `TypeVar`s)

Akuli commented:
It would be good to add a couple comments:
- Keeping `dict.fromkeys` and `OrderedDict.fromkeys` in sync (a comment in both places). Otherwise when we fix one of them, we forget to fix the other.
- Something like `# TODO: actually inherited from dict, see #issuenumberhere`

AlexWaygood commented:
> It would be good to add a couple comments:
> 
> * Keeping `dict.fromkeys` and `OrderedDict.fromkeys` in sync (a comment in both places). Otherwise when we fix one of them, we forget to fix the other.
> * Something like `# TODO: actually inherited from dict, see #issuenumberhere`

Done! I added some comments for some other `collections` methods with similar issues that I modified recently, as well.

Akuli reviewed (approved):
Thanks!
